src/config/config-helper.js

```diff
@@ -0,0 +1,158 @@
+import config from "../config/engine.json";
+import { create } from "../types/SortOption";
+
+/**
+ * This file abstracts most logic around the configuration of the Reference UI.
+ *
+ * Configuration is an important part of the "reusability" and "generic-ness" of
+ * the Reference UI, but if you are using this app as a starting point for own
+ * project, everything related to configuration can largely be thrown away. To
+ * that end, this file attempts to contain most of that logic to one place.
+ */
+
+export function getConfig() {
+  if (config.engineName) return config;
+
+  if (
+    typeof window !== "undefined" &&
+    window.appConfig &&
+    window.appConfig.engineName
+  ) {
+    return window.appConfig;
+  }
+}
+
+function toLowerCase(string) {
+  if (string) return string.toLowerCase();
+}
+
+function capitalizeFirstLetter(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+export function getTitleField() {
+  // Use "title" field as the title field if a user hasn't specified one
```

Comment:
I'm not an expert on this but since this is public-facing it might be better to not use the tone of "user".  Might be better to simply say:
```
// Use "title" field as the title field if unspecified
```